controllers/backstage_controller.go

```diff
@@ -181,22 +181,22 @@ func (r *BackstageReconciler) applyObjects(ctx context.Context, objects []model.
 		}
 
 		if err := r.patchObject(ctx, baseObject, obj); err != nil {
-			_, isStatefulSet := obj.Object().(*appsv1.StatefulSet)
-			if isStatefulSet && errors.IsInvalid(err) && errors.HasStatusCause(err, metav1.CauseTypeForbidden) {
-				// Some resources like StatefulSets allow patching a limited set of fields.
-				// That's why we are trying to delete them first, taking care of orphaning the dependents so that resources like PVCs can be retained.
-				// They will be recreated at the next reconciliation.
-				if err = r.Delete(ctx, baseObject, client.PropagationPolicy(metav1.DeletePropagationOrphan)); err != nil {
-					return fmt.Errorf("failed to delete object %s so it can be recreated: %w", obj.Object(), err)
-				}
-				lg.V(1).Info("delete object ", objDispName(obj), obj.Object().GetName())
-				continue
+			lg.V(1).Info("failed to patch object => trying to delete it so it can be recreated upon next reconciliation...",
+				objDispName(obj), obj.Object().GetName(),
+				"cause", err,
+			)
+			// Some resources like StatefulSets allow patching a limited set of fields. A FieldValueForbidden error is returned.
+			// Some other resources like Services do not support updating the primary/secondary clusterIP || ipFamily. A FieldValueInvalid error is returned.
+			// That's why we are trying to delete them first, taking care of orphaning the dependents so that they can be retained.
+			// They will be recreated at the next reconciliation.
+			// If they cannot be recreated at the next reconciliation, the expected error will be returned.
+			if err = r.Delete(ctx, baseObject, client.PropagationPolicy(metav1.DeletePropagationOrphan)); err != nil {
```

Comment:
Just realized, seems we loose one functionality here.
patchObject() also saves labels and annotations which would be lost on delete I think.
Thoughts?

Reply:
Do you mean if someone manually adds labels and annotations to the resources created by the operator? But is that case something we want to recommend?
I guess they should add such labels and annotations via the raw config instead..

Reply:
But it is probably safer in this case to revert [`e9ff462` (#384)](https://github.com/janus-idp/operator/pull/384/commits/e9ff462b926142911c9022b46af3e806a7e1f135) then, and delete the object only on known error causes. That doesn't prevent losing additional labels/annotations, but at least it limits the scope to specific errors.

Reply:
Okay, let's call this out in the release notes.

Reply:
And please add some Note that annotations and labels will be lost somewhere here https://github.com/janus-idp/operator/blob/077da8a60c37adde9a978562b7e7559932cbce02/controllers/backstage_controller.go#L196 or to the message above.

Reply:
Added in [`6d2fa79` (#384)](https://github.com/janus-idp/operator/pull/384/commits/6d2fa79077bb92c6cf6bb6a8370bb7565d696bca)